lib/TH/THHalf.h

```diff
@@ -24,17 +24,26 @@ typedef struct __thalign__(4) {
 typedef __THHalf THHalf;
 typedef __THHalf2 THHalf2;
 
-/* numeric limits */
+typedef unsigned short THHalfBits;
 
+TH_API void TH_float2halfbits(float*, THHalfBits*);
+TH_API void TH_halfbits2float(THHalfBits*, float*);
 
-TH_API THHalf TH_float2half(float a);
-TH_API float TH_half2float(THHalf a);
+TH_API THHalf TH_float2half(float);
+TH_API float  TH_half2float(THHalf);
 
 #ifndef TH_HALF_BITS_TO_LITERAL
 # define TH_HALF_BITS_TO_LITERAL(n) { n }
 #endif
 
-#define TH_HALF_MAX TH_HALF_BITS_TO_LITERAL(0x7BFF)
+#define TH_HALF_ZERO 0x0U
+#define TH_HALF_MIN  0x0400U
+#define TH_HALF_MAX  0x7BFFU
```

Comment:
why do you want all these definitions that aren't used (excluding TH_INF, which is clearly a useful addition now)?  Is there something specific you are trying to do in the future?